scripts/install/run.sh

```diff
@@ -0,0 +1,337 @@
+
+    printh "Creating assets npm package"
+    ( cd ${CLONE_DIR}/dist/assets && yarn publish --new-version=${SPARTACUS_VERSION} --registry=http://localhost:4873/ --no-git-tag-version )
+
+    printh "Creating schematics npm package"
+    ( cd ${CLONE_DIR}/projects/schematics && yarn && yarn build && yarn publish --new-version=${SPARTACUS_VERSION} --registry=http://localhost:4873/ --no-git-tag-version )
+
+    create_apps
+
+    sleep 5
+
+    (kill ${VERDACCIO_PID} || echo "Verdaccio not running on PID ${VERDACCIO_PID}. Was it already runnig before starting the script?")
+
+    npm set @spartacus:registry https://registry.npmjs.org/
+    echo "Finished: npm @spartacus:registry set back to https://registry.npmjs.org/"
+}
+
+function prestart_csr {
+    if [ -z "$CSR_PORT" ]; then
+        echo "Skipping prestart csr script"
+    else
+        printh "Prestart setup for csr app"
+        ( cd ${INSTALLATION_DIR}/csr && yarn build --prod )
+    fi
+}
+
+function start_csr_unix {
+    if [ -z "$CSR_PORT" ]; then
+        echo "Skipping csr app start"
+    else
+        prestart_csr
+        printh "Starting csr app"
+        pm2 start --name csr serve -- ${INSTALLATION_DIR}/csr/dist/csr/ --single -p ${CSR_PORT}
+    fi
+}
+
+function start_ssr_unix {
+    printh "Starting ssr app"
+    # pm2 start 
+}
+
+function start_pwa_unix {
+    printh "Starting pwa app"
+    # pm2 start serve .
+}
+
+function start_ssr_pwa_unix {
+    printh "Starting ssr with pwa app"
+    # pm2 start
+}
+
+function start_windows_apps {
+    prestart_csr
+    concurrently "serve ${INSTALLATION_DIR}/csr/dist/csr --single -p ${CSR_PORT}" --names "csr"
+}
+
+function start_apps {
+    if [[ "$OSTYPE" == "cygwin" ]]; then
+        start_windows_apps
+    elif [[ "$OSTYPE" == "msys" ]]; then
+        start_windows_apps
+    elif [[ "$OSTYPE" == "win32" ]]; then
+        start_windows_apps
+    else
+        start_csr_unix
+        # start_ssr_unix
+        # start_pwa_unix
+        # start_ssr_pwa_unix
+    fi
+}
+
+function stop_apps {
+    pm2 stop csr
+    # pm2 stop ssr
+    # pm2 stop pwa
+    # pm2 stop ssr_pwa
+}
+
+function run_e2e_tests {
+    printh "Running e2e tests on app"
+    pushd $E2E_TEST_DIR > /dev/null
+    yarn
+    popd > /dev/null
+    pushd $CLONE_DIR > /dev/null
+    yarn e2e:cy:run
+    popd > /dev/null
+}
+
+function cmd_help {
+    echo "Usage: run [command]"
+    echo "Available commands are:"
+    echo " install"
+    echo " install_npm"
+    echo " start"
+    echo " stop"
+    echo " e2e"
+    echo " help"
+}
+
+if [ -z "$1" ]; then
+    cmd_help
+    exit 1
+fi
+
+readonly commands="$1"
+
+echo "Loading configs from ./config.default.sh"
+    . ./config.default.sh
+if [ -f "./config.sh" ]; then
+    echo "Loading configs from ./config.sh"
+    . ./config.sh
+fi
+
+# top directory for the installation output (must be outside of the project)
+if [ -z $BASE_DIR ]; then
+    BASE_DIR="../../../spartacus-${SPARTACUS_VERSION}"
```

Comment:
can we extract this to a variable at the top? I am guessing we might change it soon.

Reply:
Actually, it's intended to use config file to for this. If `BASE_DIR` is set in the `config.sh` - this takes the precedence. The reason of having the variable not at the top is it has to be after config resolution (lines above).